pkg/cmd/metrics.go

```diff
@@ -3,141 +3,234 @@ package cmd
+	labelValuesCmd.Flags().StringVarP(&labelValuesStart, "start", "s", "", "Start timestamp.")
+	labelValuesCmd.Flags().StringVarP(&labelValuesEnd, "end", "e", "", "End timestamp.")
+
 	err = labelValuesCmd.MarkFlagRequired("name")
 	if err != nil {
 		panic(err)
 	}
 
+	// Rules command.
+	var ruleMatchers []string
+	var ruleType string
 	rulesCmd := &cobra.Command{
 		Use:   "rules",
 		Short: "Get rules of a tenant.",
 		Long:  "Get rules of a tenant.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			b, err := config.DoMetricsGetReq(ctx, logger, "/api/v1/rules")
+			f, currentTenant, err := fetcher.NewCustomFetcher(ctx, logger)
 			if err != nil {
-				if len(b) != 0 {
-					if perr := prettyPrintJSON(b, cmd.OutOrStdout()); perr != nil {
-						return fmt.Errorf("%v error pretty printing: %v", perr, err)
+				return fmt.Errorf("custom fetcher: %w", err)
+			}
+
+			params := &client.GetRulesParams{}
+			if len(ruleMatchers) > 0 {
+				params.Match = &ruleMatchers
+			}
+			if ruleType != "" {
+				if ruleType != "alert" && ruleType != "record" {
+					return fmt.Errorf("not valid rule type")
+				}
+				params.Type = &ruleType
+			}
+
+			resp, err := f.GetRulesWithResponse(ctx, currentTenant, params)
+			if err != nil {
+				return fmt.Errorf("getting response: %w", err)
+			}
+
+			if resp.StatusCode()/100 != 2 {
+				if len(resp.Body) != 0 {
+					if perr := prettyPrintJSON(resp.Body, cmd.OutOrStdout()); perr != nil {
+						return fmt.Errorf("request failed with statuscode %d pretty printing: %v", perr, resp.StatusCode())
 					}
-					return err
+					return fmt.Errorf("request failed with statuscode %d", resp.StatusCode())
 				}
-				return err
 			}
 
-			return prettyPrintJSON(b, cmd.OutOrStdout())
+			return prettyPrintJSON(resp.Body, cmd.OutOrStdout())
 		},
 	}
+	rulesCmd.Flags().StringArrayVarP(&ruleMatchers, "match", "m", []string{}, "Repeated series selector argument that selects the series from which to read the label values.")
+	rulesCmd.Flags().StringVarP(&ruleType, "type", "t", "", "Rule type to filter by i.e, alert or record. No filtering done if skipped.")
 
+	// Rules raw command.
 	rulesRawCmd := &cobra.Command{
 		Use:   "rules.raw",
 		Short: "Get configured rules of a tenant.",
 		Long:  "Get configured rules of a tenant.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			b, err := config.DoMetricsGetReq(ctx, logger, "/api/v1/rules/raw")
+			f, currentTenant, err := fetcher.NewCustomFetcher(ctx, logger)
 			if err != nil {
-				if len(b) != 0 {
-					fmt.Fprintln(cmd.OutOrStdout(), string(b))
-					return err
+				return fmt.Errorf("custom fetcher: %w", err)
+			}
+
+			resp, err := f.GetRawRulesWithResponse(ctx, currentTenant)
+			if err != nil {
+				return fmt.Errorf("getting response: %w", err)
+			}
+
+			if resp.StatusCode()/100 != 2 {
+				if len(resp.Body) != 0 {
+					fmt.Fprintln(cmd.OutOrStdout(), string(resp.Body))
```

Comment:
I guess this is the only non-JSON response currently, right? I wonder if it would make sense to have some pretty printing for YAML as well.

Reply:
YAML is already indented as per syntax, so don't really need to do it in this case! 🙂

Reply:
You're right, if that's already looking good, it's useless. Resolved.